src/ctelnet.cpp

```diff
@@ -1603,27 +1597,23 @@ int cTelnet::decompressBuffer( char *& in_buffer, int& length, char* out_buffer
         inflateEnd( & mZstream );
         qDebug() << "recv Z_STREAM_END, ending compression";
         this->mNeedDecompression = false;
-        this->mMCCP_version_1 = false;
-        this->mMCCP_version_2 = false;
-
-        // Reset the option state so we can re-enable compression again in the future
-        // such as in the case of a copyover -JM
-        hisOptionState[static_cast<int>(OPT_COMPRESS)] = false;
-        hisOptionState[static_cast<int>(OPT_COMPRESS2)] = false;
 
-        // To finish off this old code, here is a fix to make it stay working. -MH //
-        qDebug() << "Listening for new compression sequences or Z_OK.";
-        mWaitingForCompressedStreamToStart = true; // Was an unused boolean, thanks CR //
+        // zval should always be NULL on inflateEnd.  No need for an else block. MCCP Rev. 3 -MH //
+        initStreamDecompressor();
+        qDebug() << "Listening for new compression sequences";
+        return -1;
     }
+    /* TODO: Remove this commented code once it is deemed safe. -MH
```

Comment:
Only left it there just in case it was something you may need. To work correctly it doesn't need that and knows what's going on. So up to you on that one :)